File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    num1 = [1, 2, 3, 4, 5]
    num2 = [2, 1, 2, 3, 2, 4, 2, 5]
    num3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    prob = len(answers)
    num11 = num1 * (prob // len(num1) + 1)
    num22 = num2 * (prob // len(num2) + 1)
    num33 = num3 * (prob // len(num3) + 1)
    one = 0
    two = 0
    three = 0
    for i in range(prob):
        if num11[i] == answers[i]:
            one += 1
        if num22[i] == answers[i]:
            two += 1
        if num33[i] == answers[i]:
            three += 1
    maxi = max(one, two, three)
    answer = []
    if one == maxi:
        answer.append(1)
    if two == maxi:
        answer.append(2)
    if three == maxi:
        answer.append(3)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] first = {1, 2, 3, 4, 5};
        int[] second = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] third = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        int one = 0;
        int two = 0;
        int three = 0;
        
        for (int i = 0; i < answers.length; i++) {
            one += first[i%first.length] == answers[i] ? 1 : 0;
            two += second[i%second.length] == answers[i] ? 1 : 0;
            three += third[i%third.length] == answers[i] ? 1 : 0;
        }
        
        int max = Math.max(Math.max(one, two), three);
        
        List<Integer> lst = new ArrayList<>();
        if (one == max) lst.add(1);
        if (two == max) lst.add(2);
        if (three == max) lst.add(3);
        
        int[] answer = new int[lst.size()];
        int i = 0;
        for (int n : lst) {
            answer[i++] = n;
        }
        
        Arrays.sort(answer);
        
        return answer;       
        
    }
}